dialects/postgresql/src/main/kotlin/app/cash/sqldelight/dialects/postgresql/grammar/mixins/BindParameterMixin.kt

```diff
@@ -5,7 +5,7 @@ import com.intellij.lang.ASTNode
 
 abstract class BindParameterMixin(node: ASTNode) : BindParameterMixin(node) {
   override fun replaceWith(isAsync: Boolean, index: Int): String = when {
-    isAsync -> "$$index"
+    isAsync -> "$${index + 1}"
```

Comment:
i feel like this is a weird implementation detail, like is it always the case that `async` is what causes us to use indexed parameters instead of unindexed? I think I would prefer this function just takes a nullable index as the parameter and then the actual place where async drivers want to replace the index they can.

Reply:
and have it default to null so a bunch of callsites could be cleaned up maybe

Reply:
I have no idea what you want. I agree this case is special, but what do you mean with nullable index?

`override fun replaceWith(index: Int?): String = index?.let { "$${it + 1}" } ?: "?"`
What about postgres without async then? Or do you want to move this into the compiler? 🤔